### DISCUSSION TITLE
Can  it be use in Any java project?

### DISCUSSION THREAD

**Discussion by subham008** [eclipse/lsp4e#1125] [Q&A]
Am making my own IDE for learning, am implementing LSP.
How could I implement this client in my project and where it's documentation,

Thank you.

**Comment by mickaelistria** [comment]
LSP4E is specifically meant to intergrate with the Eclipse IDE. If you need a lower-level Java API to deal with LSP messages efficiently, please see LSP4J project. It's been used by several Java clients (this Eclipse project, NetBeans, IntelliJ) and servers (Java with JDT-LS, Microprofile, Jakarta EE, LemMinX for XML...).
HTH
